httpstan/compile.py

```diff
@@ -18,17 +18,18 @@ def compile(program_code: str, stan_model_name: str) -> str:
         ValueError: Syntax or semantic error in program code.
 
     """
-    with tempfile.NamedTemporaryFile() as fh:
-        fh.write(program_code.encode())
-        fh.flush()
-        run_args = [
-            os.path.join("httpstan", "stanc"),
-            "--name",
-            stan_model_name,
-            "--print-cpp",
-            fh.name,
-        ]
-        completed_process = subprocess.run(run_args, capture_output=True, timeout=1)
+    with importlib.resources.path(__package__, "stanc") as stanc_binary:
```

Comment:
This is interesting solution

Normally I have done similar things like

    from pathlib import Path

    path = Path(httpstan.__file__).parent.resolve() / "stanc"

Reply:
The importlib approach doesn't assume that the library is on disk. It can, in theory, deal with packages that are encapsulated in a compressed file (e.g., zipapp).